examples/conanfile/package_id/settings_and_options.rst

```diff
@@ -0,0 +1,202 @@
+                      settings
+                        os: Windows
+
+Whenever a value of the settings or subsettings changes, the pacakge ID will be different to reflect that.
+
+The most common usage for settings is to model the different project-wide aspects that might influence the package ID.
+A recipe that does that will be:
+
+.. code-block:: python
+
+    from conan import ConanFile
+
+    class Pkg(ConanFile):
+        name = "pkg"
+        version = "1.0.0"
+        settings = "os", "arch", "compiler", "build_type"
+
+Now, compiling a package with different compiler versions will result into different pacakge IDs:
+
+.. code-block:: bash
+
+    $ conan create . --settings compiler.version=192
+    ...
+    pkg/1.0.0: Package '4f267380690f99b3ef385199826c268f63147457' created
+
+    $ conan create . --settings compiler.version=193
+    ...
+    pkg/1.0.0: Package 'c13a22a41ecd72caf9e556f68b406569547e0861' created
+
+    $ conan list pkg/1.0.0:*
+    Local Cache
+      pkg
+        pkg/1.0.0
+          revisions
+            f1f48830ecb04f3b328429b390fc5de8 (2024-03-07 09:21:07 UTC)
+              packages
+                4f267380690f99b3ef385199826c268f63147457
+                  info
+                    settings
+                      arch: x86_64
+                      build_type: Release
+                      compiler: msvc
+                      compiler.cppstd: 14
+                      compiler.runtime: dynamic
+                      compiler.runtime_type: Release
+                      compiler.version: 192
+                      os: Windows
+                c13a22a41ecd72caf9e556f68b406569547e0861
+                  info
+                    settings
+                      arch: x86_64
+                      build_type: Release
+                      compiler: msvc
+                      compiler.cppstd: 14
+                      compiler.runtime: dynamic
+                      compiler.runtime_type: Release
+                      compiler.version: 193
+                      os: Windows
+
+However, there are cases when a setting should not influence the resulting package ID.
+An example of this could be when a project generates an executable binary as a compilation result. In that case, the value of the compiler
+used is not relevant for the package and we can remove its influence in the package ID like so:
+
+.. code-block:: python
+
+    from conan import ConanFile
+
+    class Pkg(ConanFile):
+        name = "pkg"
+        version = "1.0.0"
+        settings = "os", "arch", "compiler", "build_type"
+
+        def build(self):
+            # self.settings.compiler value will be used here to compile
+
+        def package_id(self):
+            # Remove compiler setting from package ID
+            del self.info.settings.compiler
+
+Why not removing the setting from the `settings` attribute? Because the compiler value is still needed in the `build()` method to perform the copilation of the executable.
```

Comment:
```suggestion
Why not removing the setting from the `settings` attribute? Because the compiler value is still needed in the `build()` method to perform the compilation of the executable.
```